src/cargo/core/compiler/mod.rs

```diff
@@ -1230,7 +1230,7 @@ fn trim_paths_args(
         // * path dependencies outside workspace root directory
         if is_local && pkg_root.strip_prefix(ws_root).is_ok() {
             remap.push(ws_root);
-            remap.push("="); // empty to remap to relative paths.
+            remap.push("=."); // explicitly remap to cwd (rustc working dir).
```

Comment:
Actually I am not sure if we want to fix this in cargo or in rustc…

Reply:
* Run `clang main.c -gdwarf -fdebug-prefix-map="$PWD"= -gsplit-dwarf`
    * on Linux `DW_AT_comp_dir` is gone, as same as the result from rustc.
    * on macOS `N_SO` and `N_OSO` are gone, as same as the result from rustc.
* Run `gcc main.c -gdwarf -fdebug-prefix-map="$PWD"= -gsplit-dwarf`
    * on Linux it generates empty `DW_AT_comp_dir`.
    * on macOS it generates `N_SO` with value `/`.

Without bothering whether this should be fixed in rustc or cargo, I would propose we add an `.` for remapping the current working directory.

For more on gcc/clang remap options, see: https://reproducible-builds.org/docs/build-path/